flat_mutation_reader.cc

```diff
@@ -107,13 +107,13 @@ flat_mutation_reader make_reversing_reader(flat_mutation_reader& original, query
     private:
         stop_iteration emit_partition() {
             auto emit_range_tombstone = [&] {
-                auto it = std::prev(_range_tombstones.end());
+                auto it = _range_tombstones.begin();
```

Comment:
Perhaps an explanation would be useful somewhere around here that `_range_tombstones` are sorted in reverse already, *because* we reverse the passed-in schema (which is done in the constructor far from here), and that's why we take from begin, not end.